app/models/post.rb

```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :user, foreign_key: 'author_id'
+  has_many :likes
+  has_many :comments, foreign_key: 'post_id'
+
+  after_save :update_post_counter
+
+  # Updates the posts counter for a user
+  def update_post_counter
+    user.increment!(:posts_counter)
+  end
+
```

Comment:
When creating `first_post`, I get the following error: 
![image](https://github.com/ShiroYaksha90/blog-app/assets/93520178/9b454106-6c7b-41d2-b909-4a6793033c2a)

This is because you reffered to the user as `author` when creating the post. 
To fix this, I suggest you replace:
``` 
 belongs_to :user, foreign_key: 'author_id'
```
with 
```
 belongs_to :author, class_name: 'User', foreign_key: 'author_id'
```


Also, in the `update_post_counter`, replace `user` with `author` (on line 10)

[OPTIONAL] Nice job `after_save` to call the `#update_post_counter` method every time a new post is saved to database. It would be even better if you  made the `update_posts_counter` method private.